tests/unit/abilities-api/wpAbilityCategory.php

```diff
@@ -0,0 +1,768 @@
+		remove_action( 'abilities_api_categories_init', $callback );
+
+		$categories = wp_get_ability_categories();
+
+		$this->assertIsArray( $categories );
+		$this->assertCount( 2, $categories );
+		$this->assertArrayHasKey( 'test-math', $categories );
+		$this->assertArrayHasKey( 'test-system', $categories );
+	}
+
+	/**
+	 * Test category is_registered method.
+	 */
+	public function test_category_is_registered(): void {
+		$this->assertFalse( $this->registry->is_registered( 'test-math' ) );
+
+		$this->register_category_during_hook(
+			'test-math',
+			array(
+				'label'       => 'Math',
+				'description' => 'Mathematical operations.',
+			)
+		);
+
+		$this->assertTrue( $this->registry->is_registered( 'test-math' ) );
+	}
+
+	/**
+	 * Test ability can only be registered with existing category.
+	 *
+	 * @expectedIncorrectUsage WP_Abilities_Registry::register
+	 */
+	public function test_ability_requires_existing_category(): void {
+		do_action( 'abilities_api_init' );
+
+		// Ensure category doesn't exist - test should fail if it does.
+		$this->assertFalse(
+			WP_Abilities_Category_Registry::get_instance()->is_registered( 'test-nonexistent' ),
+			'The test-nonexistent category should not be registered - test isolation may be broken'
+		);
+
+		// Try to register ability with non-existent category.
+		$result = wp_register_ability(
+			'test/calculator',
+			array(
+				'label'               => 'Calculator',
+				'description'         => 'Performs calculations.',
+				'category'            => 'test-nonexistent',
+				'execute_callback'    => static function () {
+					return 42;
+				},
+				'permission_callback' => '__return_true',
+			)
+		);
+
+		$this->assertNull( $result );
+		$this->assertDoingItWrongTriggered( 'WP_Abilities_Registry::register', 'not registered' );
+	}
+
+	/**
+	 * Test ability can be registered with valid category.
+	 */
+	public function test_ability_with_valid_category(): void {
+		$category_callback = static function (): void {
+			wp_register_ability_category(
+				'test-math',
+				array(
+					'label'       => 'Math',
+					'description' => 'Mathematical operations.',
+				)
+			);
+		};
+
+		add_action( 'abilities_api_categories_init', $category_callback );
+		do_action( 'abilities_api_categories_init', WP_Abilities_Category_Registry::get_instance() );
+		remove_action( 'abilities_api_categories_init', $category_callback );
+		do_action( 'abilities_api_init' );
+
+		$result = wp_register_ability(
+			'test/calculator',
+			array(
+				'label'               => 'Calculator',
+				'description'         => 'Performs calculations.',
+				'category'            => 'test-math',
+				'execute_callback'    => static function () {
+					return 42;
+				},
+				'permission_callback' => '__return_true',
+			)
+		);
+
+		$this->assertInstanceOf( WP_Ability::class, $result );
+		$this->assertSame( 'test-math', $result->get_category() );
+
+		// Cleanup.
+		wp_unregister_ability( 'test/calculator' );
```

Comment:
Nit, I would do the cleanup before assertions to ensure it always happens in case assertions fail for some reason during development.
```suggestion
		// Cleanup.
		wp_unregister_ability( 'test/calculator' );

		$this->assertInstanceOf( WP_Ability::class, $result );
		$this->assertSame( 'test-math', $result->get_category() );
```